### PR TITLE
search path variable for Cloud Sdk

### LINKED ISSUE #62: Improve Cloud SDK location detection
Use something like a `which` command to find the gcloud installation.

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #62

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java [hunk 1]
```diff
@@ -25,39 +29,70 @@
...
-        sdkDir = System.getProperty("user.home") + "/google-cloud-sdk";
-        if (!new File(sdkDir).exists()) {
-          // try devshell VM:
-          sdkDir = "/google/google-cloud-sdk";
-          if (!new File(sdkDir).exists()) {
-            // try bitnami Jenkins VM:
-            sdkDir = "/usr/local/share/google/google-cloud-sdk";
-          }
+        if (path.endsWith("google-cloud-sdk" + File.separator + "bin")) {
+          possiblePaths.add(path);
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: is this right? We need the base directory don't we?

path - bin?

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: Good catch!

**Comment by meltsufin** [comment]
@loosebazooka PTAL again

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java [hunk 2]
```diff
@@ -25,39 +29,70 @@
...
-        sdkDir = System.getProperty("user.home") + "/google-cloud-sdk";
-        if (!new File(sdkDir).exists()) {
-          // try devshell VM:
-          sdkDir = "/google/google-cloud-sdk";
-          if (!new File(sdkDir).exists()) {
-            // try bitnami Jenkins VM:
-            sdkDir = "/usr/local/share/google/google-cloud-sdk";
-          }
+        if (path.endsWith("google-cloud-sdk" + File.separator + "bin")) {
+          possiblePaths.add(path.substring(0, path.length() - 4));
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: is File.separator a string or character? would -4 actually work no matter what? should we do -3?

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java]: According to javadocs, it's a single characters. We don't need the trailing /. So, this should work.

**Comment by loosebazooka** [comment]
:ship:

**Pull request merged**